src/uct/base/uct_mem.c

```diff
@@ -101,12 +102,15 @@ ucs_status_t uct_mem_alloc(size_t length, const uct_alloc_method_t *methods,
                    params->name : "anonymous-uct_mem_alloc";
     mem_type     = (params->field_mask & UCT_MEM_ALLOC_PARAM_FIELD_MEM_TYPE) ?
                    params->mem_type : UCS_MEMORY_TYPE_HOST;
+    sys_dev      = (params->field_mask & UCT_MEM_ALLOC_PARAM_FIELD_SYS_DEVICE) ?
+                   params->sys_dev : UCS_SYS_DEVICE_ID_UNKNOWN;
```

Comment:
whould we return `INVALID_PARAM` if sys_dev is not `UCS_SYS_DEVICE_ID_UNKNOWN` and mem_type is host?

Reply:
@brminich This assumes that GPU MDs cannot allocate host memory which is true today but maybe it is better for UCT MDs to return that error than catch it here?

Reply:
if you think it can be a valid case in the future, then ok